CliFx/Services/EnvironmentVariablesProvider.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security;
+
+namespace CliFx.Services
+{
+    /// <inheritdoc />
+    public class EnvironmentVariablesProvider : IEnvironmentVariablesProvider
+    {
+        /// <summary>
+        /// Default delimiter for multiple values
+        /// </summary>
+        public const char DEFAULT_VALUES_DELIMITER = ',';
```

Comment:
Looks like Windows is using ';' as delimiter (e.g. `$env:PATH`)

Reply:
Probably should also account that ';' might be a valid character inside an environment variable... Is there some kind of convention for escaping values here?

Reply:
We can use [Path.PathSeparator](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.pathseparator?view=netstandard-2.0), looks like the env. specific separator for multiple values in an env. variable.
[According to this](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/set_1) the escape character is `^`

Reply:
Ha. Interesting. Yeah, let's use that. Let's not worry about escapes for now,
Regarding escapes, it seems that, similarly to command line arguments, environment variable enclosed in quotes is treated as one value. Maybe we should handle that?

E.g. `"value1;value2" is still treated as one value while `value1;value2` isn't. We can leave this for later.